File: app/src/main/java/com/smartsoft/pereverzev/dagger2oldtutorial/GithubApplicationComponent.java
package com.smartsoft.pereverzev.dagger2oldtutorial;

import com.smartsoft.pereverzev.dagger2oldtutorial.network.GithubService;
import com.squareup.picasso.Picasso;

import dagger.Component;

@GithubApplicationScope
@Component(modules = {GithubServiceModule.class, PicassoModule.class, ActivityModule.class})
public interface GithubApplicationComponent {

    Picasso getPicasso();

    GithubService getGithubService();
}

File: app/src/main/kotlin/com/smartsoft/pereverzev/dagger2newtutorial/GithubApplicationComponent.kt
package com.smartsoft.pereverzev.dagger2newtutorial

import com.smartsoft.pereverzev.dagger2newtutorial.network.GithubService
import com.squareup.picasso.Picasso
import dagger.Component

/**
 * Created by alexander on 03.01.2018.
 */
@GithubApplicationScope
@Component(modules = [(GithubServiceModule::class), (PicassoModule::class), (ActivityModule::class)])
interface GithubApplicationComponent {

    fun getPicasso(): Picasso

    fun getGithubService(): GithubService
}